components/github/sources/new-repository/new-repository.js

```diff
@@ -1,32 +1,41 @@
-const github = require("../../github.app.js");
+const common = require("../common-polling.js");
 
 module.exports = {
+  ...common,
   key: "github-new-repository",
   name: "New Repository",
   description: "Emit an event when a new repository is created",
-  version: "0.0.1",
-  props: {
-    github,
-    timer: {
-      type: "$.interface.timer",
-      default: {
-        intervalSeconds: 60 * 5,
-      },
+  version: "0.0.2",
+  dedupe: "last",
+  methods: {
+    generateMeta(data) {
+      const ts = new Date(data.created_at).getTime();
+      return {
+        id: data.id,
+        summary: data.full_name,
+        ts,
+      };
     },
   },
-  dedupe: "last",
   async run(event) {
+    const since = this.db.get("since");
+
     const repos = await this.github.getRepos({
-      sort: 'created',
-      direction: 'desc',
-    })
+      sort: "created",
+      direction: "desc",
+    });
+
+    let maxDate = since;
+    for (const repo of repos) {
+      if (!maxDate || new Date(repo.created_at) > new Date(maxDate)) {
+        maxDate = repo.created_at;
+      }
+      const meta = this.generateMeta(repo);
+      this.$emit(repo, meta);
+    }
```

Comment:
This check is meant to get the latest creation date among the repos returned so that it can be used as the "since" variable on the next run.

Reply:
Oops, I didn't have the `since` variable being passed into `getRepos`, so that check wasn't accomplishing anything. I updated it to pass `since` into `getRepos` so it'll only retrieve recently created or updated repos.